### DISCUSSION TITLE
Extract translation from twig variable name

### DISCUSSION THREAD

**Discussion by d47081** [symfony/symfony#52448] [Q&A] [answered]
How can I properly extract translation from variable name like `{{ category | trans }}`

That's not DB entity but the variable stored in the environment config, processed in foreach loop

```
php bin/console translation:extract --clean --domain="messages" --prefix="" --force en
```

And nothing in translation file

p.s. categories list stored in the `.env` file with `|` separator (like `category1|category2|...`) and processed to array in controller, so I need to define categories translation somewhere in controller or twig

**Comment by MatTheCat** [comment, marked as answer]
> How can I properly extract translation from variable name like `{{ category | trans }}`

If you know what values can be assigned to `category`, you can create a custom extractor by implementing the `ExtractorInterface`.

> categories list stored in the `.env` file with `|` separator (like `category1|category2|...`) and processed to array in controller

You should be using an [environment variable processor](https://symfony.com/doc/current/configuration/env_var_processors.html) like `json` or `csv`.

**Reply by d47081** [reply]
Thank you for reply, 

it's my first experience with Symfony, so I've temporarily implemented it with [custom twig filter](https://github.com/YGGverse/YGGtracker/blob/main/src/Twig/AppExtension.php#L163) but it's really strange solution - to redefine it in code :/ 

I just waited from trans method some magick but later understand that it can't extract because of not compiled yet.

So thank you for advice with `ExtractorInterface`, and I did not knew about `.env` processors so will follow your link
